controller/workitemtype.go

```diff
@@ -34,6 +45,37 @@ func (c *WorkitemtypeController) Show(ctx *app.ShowWorkitemtypeContext) error {
 		if err != nil {
 			return jsonapi.JSONErrorResponse(ctx, err)
 		}
+		// check the "If-Modified-Since header against the last update timestamp"
+		// HTTP header does not include microseconds, so we need to ignore them in the "updated_at" record field.
+		updatedAt := res.Data.Attributes.UpdatedAt.Truncate(time.Second)
+		if ctx.IfModifiedSince != nil {
+			logrus.Debug(nil, map[string]interface{}{
+				"If-Modified-Since": ctx.IfModifiedSince.UTC(),
+				"Last-Modified":     updatedAt,
+			}, "work item type conditional query")
+
+			if ctx.IfModifiedSince != nil && ctx.IfModifiedSince.UTC().After(updatedAt) {
+				return ctx.NotModified()
+			}
+		}
+		// check the ETag
+		etag := strconv.Itoa(res.Data.Attributes.Version)
+		if ctx.IfNoneMatch != nil {
+			logrus.Debug(nil, map[string]interface{}{
+				"If-None-Match": ctx.IfNoneMatch,
+				"ETag":          etag,
+			}, "work item type version")
+			if ctx.IfNoneMatch != nil && *ctx.IfNoneMatch == etag {
+				return ctx.NotModified()
+			}
+		}
+		// add the "Last-Modified" header
+		ctx.ResponseData.Header().Set("Last-Modified", updatedAt.String())
+		// add the "ETag" header
+		ctx.ResponseData.Header().Set("ETag", etag)
+		// add the "Cache-Control: max-age" header
+		ctx.ResponseData.Header().Set("Cache-Control", "max-age="+c.configuration.GetWorkItemTypeCacheControlMaxAge())
```

Comment:
I can remove that if you like, but I thought that this kind of settings should be "overridable" from the config file.